### PR TITLE
Skip NodeExpandVolume for block volumes

### LINKED ISSUE #372: NodeExpand may resize fs for block volume
Looking back at `resizer.Resize` implementation, it checks the format on the disk before attempting a resize.  The problem is, the presence of a filesystem doesn't mean that the PVC is a Filesystem type.  A user could mount the volume as Block and manage an ext4/xfs filesystem themselves.

I think fixing this may require adding VolumeCapabilities to the Controller/NodeExpandVolumeRequest

/kind bug
/assign @davidz627 
cc @gnufied

### PR DESCRIPTION
**Pull request by saad-ali**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #372

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Skip NodeExpandVolume for block volumes. 
```
/assign @msau42

### CODE REVIEW

File: pkg/gce-pd-csi-driver/node.go [hunk 1]
```diff
@@ -461,6 +461,20 @@ func (ns *GCENodeServer) NodeExpandVolume(ctx context.Context, req *csi.NodeExpa
...
 	}
 
+	volumeCapability := req.GetVolumeCapability()
+	if volumeCapability != nil {
+		// VolumeCapability is optional, if specified, validate it
+		if err := validateVolumeCapability(volumeCapability); err != nil {
+			return nil, status.Error(codes.InvalidArgument, fmt.Sprintf("VolumeCapability is invalid: %v", err))
+		}
+
+		if blk := volumeCapability.GetBlock(); blk != nil {
```
Comment (msau42) [inline pkg/gce-pd-csi-driver/node.go:471]: Can you add a unit test for this?

Comment (saad-ali) [reply pkg/gce-pd-csi-driver/node.go:471]: The `TestNodeExpandVolume` unit tests are currently broken and require more extensive work to fix: see https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/issues/573

**Comment by saad-ali** [comment]
PTAL

**Pull request merged**